### PR TITLE
Refactoring in preparation for SKARA-1146

### PR DESCRIPTION
**Pull request by erikj79**
While working on SKARA-1146, I felt that I needed to do some refactoring. To ease the review process, I've separated those changes into a separate PR. Nothing should change functionally from this change. Here is what I've done:

* Created String constants for the special custom fields we have in JBS.
* In IssueNotifier, created a method for finding the fixVersion for a branch. This code block is actually copied and sprinkled all over the code base, but for now, I will just unify the copies in the same class.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1270/head:pull/1270` \
`$ git checkout pull/1270`

Update a local copy of the PR: \
`$ git checkout pull/1270` \
`$ git pull https://git.openjdk.java.net/skara pull/1270/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1270`

View PR using the GUI difftool: \
`$ git pr show -t 1270`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1270.diff">https://git.openjdk.java.net/skara/pull/1270.diff</a>

</details>

### CODE REVIEW

File: bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java [hunk 1]
```diff
@@ -428,6 +405,23 @@ public void onNewOpenJDKTagCommits(HostedRepository repository, Repository local
         }
     }
 
+    private String getRequestedVersion(Repository localRepository, Commit commit, String tagBranch) {
```
Comment (kevinrushforth) [inline bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java:408]: Minor: would `branch` be a better name? Or is it always a tag name in this method?

File: bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/JbsBackport.java [hunk 2]
```diff
@@ -50,6 +52,7 @@ private static URI backportRequest(URI uri) {
     private Issue createBackportIssue(Issue primary) {
         var finalProperties = new HashMap<>(primary.properties());
         finalProperties.put("issuetype", JSON.of("Backport"));
+        finalProperties.remove(RESOLVED_IN_BUILD);
```
Comment (kevinrushforth) [inline bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/JbsBackport.java:55]: Minor: did you mean to make this change in this PR? It doesn't seem behavior neutral (might be related to the follow-on PR?)

**Review by kevinrushforth — APPROVED** [review]
Looks good. I left a couple questions.

File: bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java (hunk 1 above)
Comment (erikj79) [reply bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java:408]: I agree. It seems I was lazy and just went with the IDE generated extracted method signature.

File: bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/JbsBackport.java (hunk 2 above)
Comment (erikj79) [reply bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/JbsBackport.java]: Removed from this change and applied in the follow-on.

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**